paddle/cinn/common/iter_simplify.h

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2024 CINN Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <optional>
+#include <unordered_map>
+#include <vector>
+#include "paddle/cinn/common/integer_set.h"
+#include "paddle/cinn/ir/ir.h"
+#include "paddle/cinn/ir/ir_base.h"
+#include "paddle/cinn/ir/ir_mutator.h"
+
+namespace cinn {
+namespace common {
+
+bool IsIterExpr(const Expr& a, const Expr& b) {
+  return a.As<ir::IterSplit>() || a.As<ir::IterSum>() ||
+         b.As<ir::IterSplit>() || b.As<ir::IterSum>();
+}
+
+bool IsOne(const Expr& expr) {
+  if (expr.is_constant() && expr.get_constant() == 1) {
+    return true;
+  }
+  return false;
+}
+bool IsZero(const Expr& expr) {
+  if (expr.is_constant() && expr.get_constant() == 0) {
+    return true;
+  }
+  return false;
+}
+
+class IterMapToExprNormalizer : public ir::IRMutator<> {
+ public:
+  explicit IterMapToExprNormalizer(SymbolicExprAnalyzer analyzer)
```

Comment:
```suggestion
  explicit IterMapToExprNormalizer(const SymbolicExprAnalyzer& analyzer)
```
这里是值copy传入，然后引用这个copy？

Reply:
是的